contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,176 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 import "../data/DataStore.sol";
 import "../role/RoleModule.sol";
+import "../utils/BasicMulticall.sol";
 import "../fee/FeeUtils.sol";
+import "../data/Keys.sol";
+import "../v1/IVaultV1.sol";
+import "./IFeedAddress.sol";
 
 // @title FeeHandler
-contract FeeHandler is ReentrancyGuard, RoleModule {
+contract FeeHandler is ReentrancyGuard, RoleModule, BasicMulticall {
+    using SafeERC20 for IERC20;
+
     DataStore public immutable dataStore;
     EventEmitter public immutable eventEmitter;
+    address public immutable vaultV1;
+    address public immutable gmx;
 
     constructor(
         RoleStore _roleStore,
         DataStore _dataStore,
-        EventEmitter _eventEmitter
+        EventEmitter _eventEmitter,
+        address _vaultV1,
+        address _gmx
     ) RoleModule(_roleStore) {
         dataStore = _dataStore;
         eventEmitter = _eventEmitter;
+        vaultV1 = _vaultV1;
+        gmx = _gmx;
+    }
+
+    // @dev withdraw fees from this contract
+    // @param token the token for which to claim fees
+    // @param amount the amount of fees to claim
+    function withdrawFees(address token, uint256 amount) external nonReentrant onlyFeeKeeper {
```

Comment:
would be good to validate that there is a buybackBatchAmount for the "token", to prevent accidental withdrawal of fee tokens that are not GMX and not WNT

Reply:
No problem, I’ve added that validation. I had been thinking the same, but wasn't sure if we wanted the ability to withdraw tokens that weren't GMX or WNT in case of an emergency or for some other reason.